### PR TITLE
T13535

### PR DESCRIPTION
**Pull request by rmacqueen**
https://phabricator.endlessm.com/T13535

### CODE REVIEW

File: js/app/modules/filter/multi.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+/* exported Multi */
+
+const GObject = imports.gi.GObject;
+
+const Filter = imports.app.interfaces.filter;
+const HistoryStore = imports.app.historyStore;
+const Module = imports.app.interfaces.module;
+
+const Multi = new Module.Class({
```
Comment (ptomato) [inline js/app/modules/filter/multi.js]: drive-by comment, I would much rather see this implemented the same way as we do with Order: https://github.com/endlessm/eos-knowledge-lib/blob/master/js/app/interfaces/order.js#L46

**Comment by rmacqueen** [comment]
ready again

File: js/app/interfaces/filter.js [hunk 2]
```diff
@@ -60,6 +68,8 @@ const Filter = new Lang.Interface({
      */
     include: function (model) {
         let retval = this.include_impl(model);
+        if (retval && this._sub_filter)
```
Comment (ptomato) [inline js/app/interfaces/filter.js]: I don't think this is correct, because the first filter's `invert` now applies to the sub-filter. E.g. if you have `Filter.Featured(invert: true)` with `Filter.Unread(invert: false)` as the subfilter, expressing "not featured and unread", then this will also include articles that are featured and read, or not featured and read.

Comment (ptomato) [inline js/app/interfaces/filter.js]: It would be good to have tests on the `Filter` interface that would catch this.

**Comment by rmacqueen** [comment]
ready again
